modules/tinymce/src/themes/silver/main/ts/ui/general/UiFactory.ts

```diff
@@ -96,17 +100,16 @@ const interpretInForm = (parts, spec, oldBackstage) => {
   return interpretParts(parts, spec, newBackstage);
 };
 
-const interpretParts: FormPartRenderer = (parts, spec, backstage) => Obj.get(factories, spec.type).fold(
-  () => {
-    console.error(`Unknown factory type "${spec.type}", defaulting to container: `, spec);
-    return spec as AlloySpec;
-  },
-  (factory: FormPartRenderer) => factory(parts, spec, backstage)
-);
+const interpretParts = <T extends Dialog.BodyComponent>(parts: FormTypes.FormParts, spec: T, backstage: UiFactoryBackstage) =>
+  Obj.get(factories, spec.type).fold(
+    () => {
+      console.error(`Unknown factory type "${spec.type}", defaulting to container: `, spec);
+      return spec as unknown as AlloySpec;
```

Comment:
This doesn't seem correct, but at the same time it's been like that forever. I assume this is just meant to be a developer helper so left it as is for now :shrug: